src/headers/queue_linked_op.h

```diff
@@ -21,26 +21,33 @@
 
 #include <pthread.h>
 
-typedef struct linked_queue_node_t {
+typedef struct linked_queue_node_t
+{
     void *data;                       ///< pointer to the node data
     struct linked_queue_node_t *next; ///< pointer to next node
     struct linked_queue_node_t *prev; ///< pointer to prev node
 } w_linked_queue_node_t;
 
-typedef struct linked_queue_t {
-    pthread_mutex_t mutex;        ///< mutex for mutual exclusion
-    pthread_cond_t available;     ///< condition variable when queue is empty
-    unsigned int elements;        ///< counts the number of elements stored in the queue
-    w_linked_queue_node_t *first; ///< points to the first node that would go out the queue
-    w_linked_queue_node_t *last;  ///< pointer to the last node in the queue
+typedef struct linked_queue_t
+{
+    pthread_mutex_t mutex;         ///< mutex for mutual exclusion
+    pthread_cond_t available;      ///< condition variable when queue is empty
+    unsigned int elements;         ///< counts the number of elements stored in the queue
+    w_linked_queue_node_t *first;  ///< points to the first node that would go out the queue
+    w_linked_queue_node_t *last;   ///< pointer to the last node in the queue
+    void (*push_callback)(void *); ///< function pointer to an optional callback for push operation
+    void (*pop_callback)(void *);  ///< function pointer to an optional callback for pop operation
+
 } w_linked_queue_t;
 
 /**
  * @brief Initializes a new queue structure
  *
- * @return initialize queue structure
+ * @param push_callback Optional callback to run every time a push is called
+ * @param pop_callback Optional callback to run every time a pop is called
```

Comment:
Similar to the previous comment, talk about the parameter that the callback will receive.

Reply:
Done!